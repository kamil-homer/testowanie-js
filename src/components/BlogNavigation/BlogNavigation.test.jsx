import { describe, expect, test, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { BlogNavigation } from "./BlogNavigation";

describe("BlogNavigation", () => {
  vi.mock("../BlogNavigationItems/BlogNavigationItems", () => {
    return {
      BlogNavigationItems: () => <div>BlogNavigationItems</div>,
      MENU_TYPE: {},
    };
  });

  test("Should match the snapshot", () => {
    const { asFragment } = render(<BlogNavigation />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("Should display 'Hello, username' after clicking on 'Sign in' button", async () => {
    const signInLabel = "Sign in";
    const helloUserLabel = "Hello, username";
    const user = userEvent.setup();
    render(<BlogNavigation />);

    const signInButton = screen.getByText(signInLabel);
    await user.click(signInButton);
    const helloUser = await screen.findByText(helloUserLabel);

    expect(signInButton).not.toBeInTheDocument();
    expect(helloUser).toBeInTheDocument();
  });
});
