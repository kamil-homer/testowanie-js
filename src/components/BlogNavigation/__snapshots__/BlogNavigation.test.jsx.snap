// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BlogNavigation > Should match the snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-7eauvj-MuiPaper-root-MuiAppBar-root"
    style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-5c1adp-MuiContainer-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-dense css-1cxvava-MuiToolbar-root"
      >
        <div
          class="MuiBox-root css-1rw111m"
        >
          <div
            class="MuiBox-root css-1guk29"
          >
            <div>
              BlogNavigationItems
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-h10f7z"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-1ccsd5i-MuiButtonBase-root-MuiButton-root"
            id=":r0:"
            tabindex="0"
            type="button"
          >
            Sign in
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-tw6tsg-MuiButtonBase-root-MuiButton-root"
            id=":r1:"
            tabindex="0"
            type="button"
          >
            Sign up
          </button>
        </div>
        <div
          class="MuiBox-root css-zkjl6t"
        >
          <button
            aria-label="Menu button"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
            id=":r2:"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-loadingIndicator css-165vu30-MuiIconButton-loadingIndicator"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="MenuIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
